### ISSUE TITLE
'/warnaserror-:DiagnosticId' should not nullify diagnostic severity settings for 'DiagnosticId' in globalconfig

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#70570]
This is quite identical to #49446, which addressed same issue with severity setting in editorconfig file, it seems we still have the issue if the setting is specified in a globalconfig.

**Steps to Reproduce**:

1. Create a C# class library project with `net5` target framework with below code:
```csharp
class C
{
    void M() { }
}
```
2. Verify IDE suggestion/message [CA1822: Mark members as static](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822) is reported for method 'M'
3. Add a globalconfig file to the project root with following contents:
```editorconfig
is_global = true

dotnet_diagnostic.CA1822.severity = warning
```
4. Build and verify `CA1822` warning
5. Edit the csproj file to add `/warnaserror-:CA1822`:
```xml
<PropertyGroup>
  <WarningsNotAsErrors>CA1822</WarningsNotAsErrors>
</PropertyGroup>
```
6. Rebuild project

**Expected Behavior**:
`CA1822` is reported as a warning

**Actual Behavior**:
`CA1822` is not reported as a warning. `/warnaserror-:CA1822` seems to cause compiler fallback `CA1822` to its default severity `Info`/`Suggestion` and ignores globalconfig entry that promotes it to a warning.
